include/dictionary.h

```diff
@@ -90,5 +90,21 @@ int dict_add(dict_t *d, char *str);
  */
 int dict_read(dict_t *d, char *file);
 
+/*
+ * Returns the n closest words to a given string in a dictionary.
+ * 
+ * Parameters:
+ *  - d: A dictionary. Must point to a dictionary allocated with dict_new
+ *  - str: A string. This will be the (misspelled) word to match
+ *  - max_edits: the maximum levenshtein distance the words in the set can have
+ *  - n: the number of strings to return. 
+ * 
+ * Returns:
+ *  - The first n strings with the smallest distance, where ties are broken by alphabetical order.
+ *    If there aren't enough matching strings, each remaining spot is set to NULL.
+ *  - NULL if there was an error
```

Comment:
Can I just ask for confirmation that the last thing i.e. `results[n]` will be NULL?

Reply:
results[n] doesn't exist.

Calling dict_suggestions with n = 5 but only 3 matches will return:
```match_1, match_2, match_3, NULL, NULL```

Calling dict_suggestions with n = 5 but having 7 matches will return:
```match_1, match_2, match_3, match_4, match_5```
and the matches will be sorted by edit_distance from the original word and ties broken by alphabetic order.

Reply:
Understood!